dbt/include/snowflake/macros/adapters.sql

```diff
@@ -150,21 +150,83 @@
 {% endmacro %}
 
 
-{% macro snowflake__list_relations_without_caching(schema_relation) %}
+{% macro snowflake__list_relations_without_caching(schema_relation, max_iter=10, max_results_per_iter=10000) %}
+
+  {%- set max_total_n = max_results_per_iter * max_iter -%}
+
   {%- set sql -%}
-    show terse objects in {{ schema_relation }}
+    show terse objects in {{ schema_relation }} limit {{ max_results_per_iter }}
   {%- endset -%}
 
   {%- set result = run_query(sql) -%}
-  {% set maximum = 10000 %}
-  {% if (result | length) >= maximum %}
-    {% set msg %}
-      Too many objects in schema  {{ schema_relation }}! dbt can only get
-      information about schemas with fewer than {{ maximum }} objects.
-    {% endset %}
-    {% do exceptions.raise_compiler_error(msg) %}
+  {%- set n = (result | length) -%}
+  {%- set counts_array = [n] -%}
+  {%- set paginated_results_array = [] -%}
+
+  {% if n >= max_results_per_iter %}
+
+    {# cant update a variable in the context of a loop, so sticking in an array instead #}
+    {% set last_values_array = [] %}
+
+    {# get the last value from the name column #}
+    {% do last_values_array.append(result.columns[1].values()[-1]) %}
+
+    {% for _ in range(1, (max_iter + 1) ) %}
+
+      {#
+        terminating condition: At some point the user needs to be reasonable with how
+        many objects are contained in their schemas
+      #}
+
+      {%- if loop.index == max_iter -%}
+
```

Comment:
nit: nix some of the whitespace for this if block through 189

Reply:
done